Toshi/Source/Toshi/File/TFile.h

```diff
@@ -18,16 +20,25 @@ namespace Toshi
 		virtual ~TFileSystem();
 
 		// Override these funcs in TNativeFileSystem
+		virtual TFile* CreateFile(TCString const& fn, uint32_t flags) = 0;
+		virtual void DestroyFile(TFile*) = 0;
 		virtual bool RemoveFile(TCString const&) { return true; }
+		virtual TCString MakeInternalPath(TCString const&) = 0;
 		virtual bool GetFirstFile(TCString const&, TCString&, unsigned int) { return false; }
-		virtual bool GetNextFile(TCString&, unsigned int) { return false; }
+		virtual bool GetNextFile(TCString&, uint32_t) { return false; }
 		virtual void SetPrefix(TCString const&);
+		virtual bool MakeDirectory(TCString const&) = 0;
 
 		inline TCString const& GetName() const { return m_sName; }
 		inline TCString const& GetPrefix() const { return m_prefix; }
 
 		TFileSystem& operator=(TFileSystem& a_rFileSystem);
 
+	public:
+		// Creates native file system based on the used platform
+		static STL::Ref<TFileSystem> CreateNative(const char* name);
```

Comment:
the main goal of TNativeFile and TNativeFileSystem is to make support of multiplatform easy so the CreateNative function will create an instance of a needed class based on the building platform automatically instead of creating TNativeFileSystem manually

Reply:
Ok but was this also used in the orginial or is that new?

Reply:
Maybe they used an inline function to do that or maybe they had different versions of Toshi's source code for different platforms and had to modify each one manually. They could also have used a macros to do the job but that means TNativeFile had code for every single platform in a single cpp file which makes the cpp file very big and hard to read